generator.rb

```diff
@@ -0,0 +1,85 @@
+def is_vowel?(letter)
+    letter.each_char do |char|
+        if char == "a" || char == "e" || char == "i" || char == "o" || char == "u"
+            return true
+        else
+            return false
+        end
+    end
+end
+
+def input_validation(word)
+    while word.length < 2 do
+        puts "The word must have 2 or more letters. Please enter again."
+        word = gets.chomp
+    end
+    
+    while !(word.include?('a') || word.include?('e') || word.include?('i') || word.include?('o') || word.include?('u')) do
+        puts "Input is invalid. The word needs at least 1 vowel. Try again."
+        word = gets.chomp
+    end
+    return word
+end
+
+def run_generator
+    puts "What is the first word to combine?"
+    first_word = input_validation(gets.chomp)
+
+    puts "What is the second word to combine?"
+    second_word = input_validation(gets.chomp)
+    
+    puts "\nThe first input is #{first_word}"
+    puts "The second input is #{second_word}"
+
+    i1 = -1 # variable to store the index of the last vowel of the 1st word
+    tic1 = -1 # initialize variable to temporarily count indexes to add only if a vowel is found
```

Comment:
I don't know what the variable name `tic1` means!

Reply:
Changed variable name to better reflect its purpose.